reader-shp/src/test/java/com/graphhopper/reader/shp/ShapeFileReaderTest.java

```diff
@@ -0,0 +1,289 @@
+import com.graphhopper.GraphHopper;
+import com.graphhopper.PathWrapper;
+import com.graphhopper.reader.osm.GraphHopperOSM;
+import com.graphhopper.routing.util.CarFlagEncoder;
+import com.graphhopper.routing.util.EncodingManager;
+import com.graphhopper.util.DistanceCalc;
+import com.graphhopper.util.Helper;
+import com.graphhopper.util.PointList;
+import com.graphhopper.util.shapes.GHPoint;
+
+/**
+
+ * @author Philip Welch
+ */
+public class ShapeFileReaderTest {
+
+	private static final String shapefile = "/data/gis.osm_roads_free_1.shp";
+	private static final String pbf = "/data/malta-latest.osm.pbf";
+	private static final String tempOutputDirFromShp = "target/test-db-shp";
+	private static final String tempOutputDirFromPbf = "target/test-db-pbf";
+	private static GraphHopper hopperShp;
+	private static GraphHopper hopperPbf;
+    private final DistanceCalc distCalc = Helper.DIST_EARTH;
+    private static Exception BEFORE_CLASS_EXCEPTION=null;
+
+	private static class FromToPair{
+		final GHPoint from;
+		final GHPoint to;	
+		
+		FromToPair(double fromLat, double fromLng, double toLat, double toLng) {
+			this( new GHPoint(fromLat, fromLng), new GHPoint(toLat, toLng));
+		}
+
+		FromToPair(GHPoint from, GHPoint to) {
+			this.from = from;
+			this.to = to;
+		}
+
+		
+		PathWrapper getPath( GraphHopper hopper, boolean assertNoErrors) {
+			GHRequest request = new GHRequest(from, to).setVehicle("car");
+			GHResponse response = hopper.route(request);
+			
+			if(assertNoErrors){
+				assertFalse(response.hasErrors());				
+			}
+	
+			if(!response.hasErrors()){
+				return response.getBest();
+			}
+			return null;
+		}
+		
+	}
+	
+	private static class ExpectedDuration extends FromToPair {
+		final double minSecs;
+		final double maxSecs;
+
+		private ExpectedDuration(double fromLat, double fromLng, double toLat, double toLng, double minSecs, double maxSecs) {
+			super(fromLat, fromLng, toLat, toLng);
+			this.minSecs = minSecs;
+			this.maxSecs = maxSecs;
+		}
+	}
+
+	private static GraphHopper initHopper(GraphHopper gh, String inputFile, String outDir) {
+		URL resourceURL = ShapeFileReaderTest.class.getResource(inputFile);
+		try {
+			inputFile = new File(resourceURL.toURI()).getAbsolutePath();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+
+		// turn off geometry simplification so geometry should be the same between pbf and shapefile readers
+		gh.setWayPointMaxDistance(0);
+		return gh.setStoreOnFlush(false).setDataReaderFile(inputFile).setGraphHopperLocation(new File(outDir).getAbsolutePath())
+				.setEncodingManager(new EncodingManager(new CarFlagEncoder())).setCHEnabled(false).importOrLoad();
+
+	}
+
+	/**
+	 * Build the graphs once only for the various tests
+	 */
+	@BeforeClass
+	public static void setupBeforeClass() {
+		try {
+			new File(tempOutputDirFromShp).mkdirs();
+			new File(tempOutputDirFromPbf).mkdirs();
+			
+			System.out.println("Building from shapefile");
```

Comment:
Prefer to keep the printlns if OK. There's some console output from the tests I like to check : )